src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -201,6 +202,8 @@ export class PersistentTerminalProcess extends Disposable {
 	readonly onProcessReady = this._onProcessReady.event;
 	private readonly _onProcessTitleChanged = this._register(new Emitter<string>());
 	readonly onProcessTitleChanged = this._onProcessTitleChanged.event;
+	private readonly _onBufferData = this._register(new Emitter<string>());
+	readonly onBufferData = this._onBufferData.event;
```

Comment:
I don't think we need this event, instead we will write the joined string to `this._onProcessData`. Also the regular handling of process data will need to get removed and replaced with `persistemTerminalProcess.onProcessData(...)`

https://github.com/microsoft/vscode/blob/e70d90fddf3916edb4aabd27e377dc1d95bcba43/src/vs/platform/terminal/node/ptyService.ts#L68